packages/grid/src/Grid.tsx

```diff
@@ -817,6 +817,19 @@ class Grid extends PureComponent<GridProps, GridState> {
     const metricState = this.getMetricState(state);
     this.metrics = metricCalculator.getMetrics(metricState);
 
+    // Fix for https://github.com/deephaven/web-client-ui/issues/936
+    // If metrics update from something like visibility panel, the lastLeft
+    // Can change due to hidden columns. Left should always be <= lastLeft
+    // Same for top
+    const { left, top } = state;
+    const { lastLeft, lastTop } = this.metrics;
+    if (left > lastLeft) {
+      this.setState({ left: lastLeft });
+    }
+    if (top > lastTop) {
+      this.setState({ top: lastTop });
+    }
```

Comment:
You also need to set `leftOffset` and `topOffset` when resetting the scroll position.
```suggestion
    const { left, top } = state;
    const { lastLeft, lastTop } = this.metrics;
    if (left > lastLeft) {
      this.setState({ left: lastLeft, leftOffset: 0 });
    }
    if (top > lastTop) {
      this.setState({ top: lastTop, topOffset: 0 });
    }
```

We do this in `handleResize` as well, consider refactoring it into it's own `checkScrollBounds` method or something instead.